docker/transformers-pytorch-amd-gpu/Dockerfile

```diff
@@ -22,7 +22,11 @@ RUN python3 -m pip install --no-cache-dir --upgrade pip setuptools ninja git+htt
 
 ARG REF=main
 WORKDIR /
+
+# Invalidate docker cache from here if new commit is available.
+ADD https://api.github.com/repos/huggingface/transformers/git/refs/heads/main version.json
```

Comment:
Is this necessary? I mean, do you see any issue so adding this line to avoid it?

Reply:
For the motivation: https://stackoverflow.com/questions/36996046/how-to-prevent-dockerfile-caching-git-clone

We do not want docker to cache the following `git clone`.

Reply:
I am not familiar with this part. Is it relevant only if we build the image on the same machine multiple times?
So far we don't have such issue, but maybe that is because we build the image on Github Actions hosted machine rather than self-hosted VM.

Reply:
I had the issue locally when using this docker image, where I would not pick up the latest transformers commit due to docker cache. I think it is useful to make sure we use the latest commit - even though in the CI this is not an issue.

Reply:
ok。

So the issue occurs when you run the docker image, not at the time of docker build, right?

Reply:
If this is more on when we use the image, I think it is expected. These images are not built for long-term usage: they are re-built on a **daily basis** to run the CI at **a specific and the same commit** .

Anyone wants to use the image locally is responsible to do `git fetch && git pull` (or `git checkout`).

If we accept this change (and IIUC), it means, on the CI (GitHub Actions),  each job may get different commits to run the test against, which is not what we want.

The above is just to explain the current behavior (before this change), not meaning we have issue, as on CI, we have

```
git fetch && git checkout ${{ github.sha }}
```
so we are safe.

Reply:
> So the issue occurs when you run the docker image, not at the time of docker build, right?

No, `docker build` caches intermediate layers, one of them being `git clone https://github.com/huggingface/transformers && cd transformers && git checkout $REF`. If we use `REF=main` and actually want to use the latest commit, we need to invalidate the docker cache and this is what this `ADD` is doing.

Otherwise, docker cached intermediate layer is used and we may use an outdated commit compared to the latest commit available at build time.

Reply:
OK, so during build time. Thanks for the detailed explanation.